packages/connectors-web/src/alchemySmartWallet.ts

```diff
@@ -1,85 +1,333 @@
+    // `config` contains an event emitter that all of Wagmi listens to. Don't
+    // let the inner connector get its hands on it: we want to fully encapsulate
+    // the inner connector so it doesn't affect global state. Note that the uid
+    // doesn't matter since this emitter is internal to this connector instance.
+    const innerEmitter = new Emitter("inner-uid");
+    const innerConnector = options.ownerConnector({
+      ...config,
+      emitter: innerEmitter,
+    });
+
+    /*
+     * We'll return this useless provider when the smart wallet connector is not
+     * connected. We need to return _something_ in response to `getProvider`
+     * even when not connected, or Wagmi will skip over this connector.
+     *
+     * TODO: Should we return the same provider instance every time, with that
+     * instance being a proxy object routing to either this or the real provider
+     * if it exists? We would need to do that if Wagmi held on to a provider
+     * that it first got during initialization. It doesn't seem to do that
+     * today, but what if it changed behavior?
+     */
+    const loggedOutProvider: Provider = {
+      on: () => {},
+      removeListener: () => {},
+      request: async (requestParams) => {
+        if (requestParams.method === "eth_accounts") {
+          return [] as any;
+        }
+        const baseProvider = (await innerConnector.getProvider()) as any;
+        return baseProvider.request(requestParams);
+      },
+    };
+
+    let smartWalletClientPromise: Promise<SmartWalletClient> | undefined;
+    let currentChainId: number | undefined;
+    let currentInnerAccount: Address | undefined;
+
+    const resetState = (): void => {
+      smartWalletClientPromise = undefined;
+      currentChainId = undefined;
+      currentInnerAccount = undefined;
+    };
+
+    const createClient = async (): Promise<SmartWalletClient> => {
+      const chainId = await innerConnector.getChainId();
+      const innerAccounts = await innerConnector.getAccounts();
+      if (innerAccounts.length === 0) {
+        throw new Error("No accounts found in base connector");
+      }
+      // TODO: do we always take the first account?
+      const signerAddress = innerAccounts[0];
+      const chain = getChainFromConfig(config, chainId);
+      const provider = (await innerConnector.getProvider()) as EIP1193Provider;
+      const signer = createWalletClient({
+        account: signerAddress,
+        chain,
+        transport: custom(provider),
+      });
+      const smartWalletClient = createSmartWalletClient({
+        signer,
+        transport: alchemyTransport({
+          apiKey: options.apiKey,
+          jwt: options.jwt,
+          url: options.url ?? "https://api.g.alchemy.com/v2",
+        }),
+        chain,
+        policyId: options.policyId,
+        policyIds: options.policyIds,
+      });
+      // TODO: Calling `client.getProvider()` has the side effect of
```

Comment:
we should probably remove this from the provider soon. this is also kinda preventing us from specifying account options (i.e. 7702) here.

Reply:
Oh you're right. Let me think about what to do with that.